torrentool/torrent2.py

```diff
@@ -0,0 +1,444 @@
+from os.path import join, isdir, getsize, normpath, basename, getsize
+from os import walk, sep
+from hashlib import sha1
+from datetime import datetime
+from calendar import timegm
+from functools import reduce
+import math
+
+from .bencode import Bencode
+from .exceptions import TorrentError
+from .utils import get_app_version
+
+
+class Torrent(object):
+    """Represents a torrent file, and exposes utilities to work with it."""
+
+    _filepath = None
+
+    def __init__(self,
```

Comment:
Common approach is to encapsulate settings in some settings object, and pass that object, yet, here I hardy can see any reasons for a separate object, since most of settings do belong solely to size calculation.